test/app-luatest/gh_8022_tarantool_build_flags_test.lua

```diff
@@ -0,0 +1,10 @@
+local t = require('luatest')
+
+local g = t.group('gh-8022')
+
+-- Check that `tarantool.build.flags` contains an optimization level setting.
+-- It means that config-specific flags are included into the variable.
+g.test_build_target = function()
+    local tarantool = require('tarantool')
+    t.assert_str_contains(tarantool.build.flags, '-O')
```

Comment:
This is a user-visible change. I think we need a changelog for it.

Reply:
Done.